### PR TITLE
Task 75 contractor profile implementation

### PR DESCRIPTION
**Pull request by s1hu**
Implemented contractor profile. Pls remove .env, ignore changes in package.json

### CODE REVIEW

**Review by and-le — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts and ensure that the linter and tests are passing.

File: backend/controllers/contractorController.js [hunk 1]
```diff
@@ -86,8 +87,9 @@ const updateContractorProfile = async (req, res) => {
     try {
         const contractorId = req.user._id;
         // Get the fields we want to update
-        const { description, phone_number, city, state, zip_code, website_url, services } = req.body;
+        const { description, phone_number, city, state, zip_code, website_url, services, profile_name } = req.body;
```
Comment (and-le) [inline backend/controllers/contractorController.js:90]: The `profile_name` field can use the existing `name` field. There is no need to create a new field for this. Please replace the `profile_name` with name.

File: backend/controllers/contractorController.js [hunk 2]
```diff
@@ -86,8 +87,9 @@ const updateContractorProfile = async (req, res) => {
...
         const updateQuery = {
+            profile_name: profile_name,
```
Comment (and-le) [inline backend/controllers/contractorController.js:92]: Replace `profile_name` with `name`.

File: backend/models/contractorModel.js [hunk 3]
```diff
@@ -44,6 +44,11 @@ const contractorSchema = new Schema({
     required: false,
     default: ''
   },
+  profile_name: {
```
Comment (and-le) [inline backend/models/contractorModel.js:47]: `profile_name` is already being represented by `name`. Or do you intend on using the fields in different ways?

File: client/src/components/dashboard/contractor/Profile/ProfileEdit.jsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        const getProfile = async () => {
+            setIsLoading(true);
+            const response = await fetch('api/contractor/profile', {
+                headers:{'Authorization': `Bearer ${user.token}`}
+            });
+
+            const json = await response.json();
+            console.log(user.token);
+            console.log(json);
+            setProfileName(json.profile_name? json.profile_name:'');
```
Comment (and-le) [inline client/src/components/dashboard/contractor/Profile/ProfileEdit.jsx:29]: See other comments about `profile_name` versus `name`.

**Review by and-le — CHANGES_REQUESTED** [review]
Please see the discussion no using `profile_name` versus `name`.

File: backend/controllers/contractorController.js (hunk 1 above)
Comment (s1hu) [reply backend/controllers/contractorController.js:90]: profile_name and name are two different fields with name being the contractor's person's name and profile_name being the production name shown on listing

**Review by s1hu — COMMENTED** [review]
profile_name issue is another necessary field, resolved

**Review by and-le — APPROVED** [review]
Approved!

**Pull request merged**